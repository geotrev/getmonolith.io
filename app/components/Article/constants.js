import { Tooltips, Accordions, Collapsibles, Modals, Dropdowns } from "undernet"

export const COMPONENTS = [Tooltips, Accordions, Collapsibles, Modals, Dropdowns]
