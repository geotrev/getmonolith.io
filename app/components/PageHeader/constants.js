export const FOCUSABLE_TABINDEX = "-1"
export const UNFOCUSABLE_TABINDEX = null
