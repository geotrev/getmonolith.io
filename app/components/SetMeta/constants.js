export const titleTemplate = placeholder => `Undernet â€“ ${placeholder}`
