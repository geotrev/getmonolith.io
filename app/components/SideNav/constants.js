import { docPages } from "app/routes"

const {
  introduction,
  download,
  css,
  javascript,
  grid,
  typography,
  buttons,
  forms,
  modals,
  collapsibles,
  accordions,
  dropdowns,
  tooltips,
  alignment,
  offsetOrder,
  text,
  display,
  color,
  spacing,
} = docPages

export const NAV_DATA = [
  {
    header: "Overview",
    links: [
      { name: "Introduction", url: introduction },
      { name: "Download", url: download },
      { name: "CSS", url: css },
      { name: "JavaScript", url: javascript },
    ],
  },
  {
    header: "Layout",
    links: [
      { name: "Flex Grid", url: grid },
      { name: "Alignment", url: alignment },
      { name: "Offset / Order", url: offsetOrder },
      { name: "Spacing", url: spacing },
    ],
  },
  {
    header: "Elements",
    links: [
      { name: "Buttons", url: buttons },
      { name: "Forms", url: forms },
      { name: "Typography", url: typography },
    ],
  },
  {
    header: "Components",
    links: [
      { name: "Collapsibles", url: collapsibles },
      { name: "Accordions", url: accordions },
      { name: "Dropdowns", url: dropdowns },
      { name: "Modals", url: modals },
      { name: "Tooltips", url: tooltips },
    ],
  },
  {
    header: "Utilities",
    links: [
      { name: "Color", url: color },
      { name: "Display", url: display },
      { name: "Text", url: text },
    ],
  },
]
