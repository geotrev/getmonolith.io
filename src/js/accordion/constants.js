export const Selectors = {
  // unique
  DATA_ACCORDION: "data-accordion",
  DATA_COLLAPSIBLE: "data-collapsible",
  DATA_TOGGLE_MULTIPLE: "data-toggle-multiple",
  // common
  DATA_PARENT: "data-parent",
  DATA_VISIBLE: "data-visible",
  DATA_TARGET: "data-target",
  // accessibility
  ARIA_EXPANDED: "aria-expanded",
  ARIA_HIDDEN: "aria-hidden",
}

export const Messages = {
  NO_ACCORDION_ID_ERROR:
    "Could not initialize accordion; you must include a value for the 'data-accordion' attribute.",
  NO_ACCORDION_ERROR: id => `Could not find element matching [data-accordion='${id}']`,
  TRIGGERS_TO_COLLAPSIBLES_LENGTH_WARNING: (id, triggersLength, collapsiblesLength) =>
    `Your accordion with id '${id}' has ${triggersLength} triggers and ${collapsiblesLength} collapsibles; make sure there is a trigger for each collapsible!`,
}
