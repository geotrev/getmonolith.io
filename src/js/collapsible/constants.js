export const Selectors = {
  // unique
  DATA_COLLAPSIBLE: "data-collapsible",
  // common
  DATA_VISIBLE: "data-visible",
  DATA_TARGET: "data-target",
  DATA_PARENT: "data-parent",
  // accessibility
  ARIA_EXPANDED: "aria-expanded",
  ARIA_CONTROLS: "aria-controls",
  ARIA_HIDDEN: "aria-hidden",
  ARIA_LABELLEDBY: "aria-labelledby",
  TABINDEX: "tabindex",
  // classes
  TRIGGER_READY_CLASS: "collapsible-trigger-ready",
  CONTENT_READY_CLASS: "collapsible-content-ready",
}

export const CssProperties = {
  HEIGHT: "height",
  VISIBILITY: "visibility",
}

export const CssValues = {
  AUTO: "auto",
  HIDDEN: "hidden",
  VISIBLE: "visible",
}

export const Events = {
  CLICK: "click",
  TRANSITIONEND: "transitionend",
}

export const Messages = {
  NO_COLLAPSIBLE_ID_ERROR:
    "Could not initialize collapsible; you must include a value for the 'data-collapsible' attribute.",
  NO_TRIGGER_ERROR: id =>
    `Could not find collapsible trigger with [data-target='${id}']; you can't have a collapsible without a trigger.`,
  NO_TRIGGER_ID_ERROR: id => `Could not find id on collapsible trigger with [data-target='${id}'].`,
  NO_CONTENT_ERROR: id =>
    `Could not find collapsible content with id '${id}'; you can't have a collapsible without content.`,
}
