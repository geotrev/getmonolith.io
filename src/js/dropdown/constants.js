export const KeyCodes = {
  TAB: 9,
  SHIFT: 16,
  ESCAPE: 27,
  ARROW_UP: 38,
  ARROW_DOWN: 40,
}

export const Selectors = {
  // unique
  DATA_DROPDOWN: "data-dropdown",
  DROPDOWN_MENU_CLASS: "dropdown-menu",
  // common
  DATA_TARGET: "data-target",
  DATA_PARENT: "data-parent",
  DATA_VISIBLE: "data-visible",
  // accessibility
  TABINDEX: "tabindex",
  ARIA_HASPOPUP: "aria-haspopup",
  ARIA_CONTROLS: "aria-controls",
  ARIA_LABELLEDBY: "aria-labelledby",
  ARIA_EXPANDED: "aria-expanded",
  // classes
  OVERLAY_OPEN: "overlay-open",
}

export const Events = {
  KEYDOWN: "keydown",
  CLICK: "click",
  BLUR: "blur",
}

export const Messages = {
  NO_DROPDOWN_ID_ERROR:
    "Could not setup dropdown. Make sure it has a valid [data-dropdown] attribute with a unique id as its value.",
  NO_MENU_ERROR: attr => `Could not find menu associated with ${attr}.`,
  NO_DROPDOWN_ITEMS_ERROR: attr => `Could not find any list items associated with ${attr}.`,
  NO_DROPDOWN_ACTIONS_ERROR: attr =>
    `Could not find any button or anchor elements associated with ${attr}.`,
  NO_PARENT_ERROR: "Could not find dropdown button's [data-parent] attribute.",
}
