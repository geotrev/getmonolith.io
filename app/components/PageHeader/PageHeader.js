import React, { createRef, useState, useEffect } from "react"
import { withLastLocation } from "react-router-last-location"
import PropTypes from "prop-types"

import { FOCUSABLE_TABINDEX, UNFOCUSABLE_TABINDEX } from "./constants"

const PageHeader = props => {
  const [tabIndex, setTabIndex] = useState(UNFOCUSABLE_TABINDEX)
  const headerRef = createRef()

  useEffect(() => {
    if (tabIndex === FOCUSABLE_TABINDEX) {
      headerRef.current.focus()
    }
  }, [tabIndex])

  useEffect(() => {
    if (props.lastLocation) setTabIndex(FOCUSABLE_TABINDEX)
  }, [])

  const handleBlur = () => {
    if (tabIndex === UNFOCUSABLE_TABINDEX) return
    setTabIndex(UNFOCUSABLE_TABINDEX)
  }

  return (
    <h1 onBlur={handleBlur} tabIndex={tabIndex} ref={headerRef} className={props.className}>
      {props.children}
    </h1>
  )
}

PageHeader.propTypes = {
  children: PropTypes.any.isRequired,
  className: PropTypes.string,
  history: PropTypes.object,
  lastLocation: PropTypes.object,
}

export default withLastLocation(PageHeader)
