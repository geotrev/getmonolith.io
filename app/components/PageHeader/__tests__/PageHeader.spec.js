import React from "react"
import PageHeader from "../PageHeader"
import { LastLocationProvider } from "react-router-last-location"
import { BrowserRouter as Router } from "react-router-dom"

import { UNFOCUSABLE_TABINDEX } from "../constants"

const baseProps = {
  className: "test",
}

const mountComponent = newProps => {
  return mount(
    <Router>
      <LastLocationProvider>
        <PageHeader {...Object.assign({}, baseProps, newProps)}>{"Test Header"}</PageHeader>
      </LastLocationProvider>
    </Router>
  )
}

describe("<PageHeader />", () => {
  describe("#render", () => {
    it("renders", () => {
      const wrapper = mountComponent()
      expect(wrapper).toMatchSnapshot()
    })
  })

  describe("#state", () => {
    it("does not focus header if lastLocation prop is not passed", () => {
      const wrapper = mountComponent()
      const headerFocused = wrapper.find("h1").is(":focus")
      expect(headerFocused).toBe(false)
    })

    it("focuses header if lastLocation prop is passed", () => {
      const wrapper = mountComponent({ lastLocation: { location: "/new-location" } })
      const headerFocused = wrapper.find("h1").is(":focus")
      expect(headerFocused).toBe(true)
    })

    it("resets h1 tabIndex to null when blurred", () => {
      const wrapper = mountComponent({ lastLocation: { location: "/new-location" } })
      wrapper.find("h1").simulate("blur")
      expect(wrapper.find("h1").props().tabIndex).toEqual(UNFOCUSABLE_TABINDEX)
    })
  })
})
