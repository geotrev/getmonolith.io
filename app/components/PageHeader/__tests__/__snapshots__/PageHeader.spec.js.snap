// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageHeader /> #render renders 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <withRouter(LastLocationProvider)>
      <LastLocationProvider
        history={
          Object {
            "action": "POP",
            "block": [Function],
            "createHref": [Function],
            "go": [Function],
            "goBack": [Function],
            "goForward": [Function],
            "length": 1,
            "listen": [Function],
            "location": Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
            "push": [Function],
            "replace": [Function],
          }
        }
        location={
          Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          }
        }
        match={
          Object {
            "isExact": true,
            "params": Object {},
            "path": "/",
            "url": "/",
          }
        }
        watchOnlyPathname={false}
      >
        <withRouter(WithLastLocation(PageHeader))
          className="test"
        >
          <WithLastLocation(PageHeader)
            className="test"
            history={
              Object {
                "action": "POP",
                "block": [Function],
                "createHref": [Function],
                "go": [Function],
                "goBack": [Function],
                "goForward": [Function],
                "length": 1,
                "listen": [Function],
                "location": Object {
                  "hash": "",
                  "pathname": "/",
                  "search": "",
                  "state": undefined,
                },
                "push": [Function],
                "replace": [Function],
              }
            }
            location={
              Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": undefined,
              }
            }
            match={
              Object {
                "isExact": true,
                "params": Object {},
                "path": "/",
                "url": "/",
              }
            }
          >
            <PageHeader
              className="test"
              history={
                Object {
                  "action": "POP",
                  "block": [Function],
                  "createHref": [Function],
                  "go": [Function],
                  "goBack": [Function],
                  "goForward": [Function],
                  "length": 1,
                  "listen": [Function],
                  "location": Object {
                    "hash": "",
                    "pathname": "/",
                    "search": "",
                    "state": undefined,
                  },
                  "push": [Function],
                  "replace": [Function],
                }
              }
              lastLocation={null}
              location={
                Object {
                  "hash": "",
                  "pathname": "/",
                  "search": "",
                  "state": undefined,
                }
              }
              match={
                Object {
                  "isExact": true,
                  "params": Object {},
                  "path": "/",
                  "url": "/",
                }
              }
            >
              <h1
                className="test"
                onBlur={[Function]}
                tabIndex={null}
              >
                Test Header
              </h1>
            </PageHeader>
          </WithLastLocation(PageHeader)>
        </withRouter(WithLastLocation(PageHeader))>
      </LastLocationProvider>
    </withRouter(LastLocationProvider)>
  </Router>
</BrowserRouter>
`;
